src/main/java/uk/gov/register/serialization/RSFExecutor.java

```diff
@@ -70,22 +72,23 @@ private RegisterResult validate(RegisterCommand command, Register register, int
     }
 
     private Boolean validateAppendEntry(RegisterCommand command, Register register, Map<HashValue, Integer> hashRefLine) {
-        String entrySha256 = command.getCommandArguments().get(1);
-        HashValue hashValue = HashValue.decode(HashingAlgorithm.SHA256, entrySha256);
-        if (hashRefLine.containsKey(hashValue)) {
-            hashRefLine.put(hashValue, 0);
-        } else {
-            Optional<Item> item = register.getItemBySha256(hashValue);
-            if (!item.isPresent()) {
-                return true;
+        String delimitedHashes = command.getCommandArguments().get(1);
+        List<HashValue> hashes = Splitter.on(";").splitToList(delimitedHashes).stream()
+                .map(s -> HashValue.decode(SHA256, s)).collect(toList());
+
+        hashes.stream().forEach(hashValue -> {
+            if (hashRefLine.containsKey(hashValue)) {
+                hashRefLine.put(hashValue, 0);
             }
-        }
-        return false;
+        });
+
+        return hashes.stream().allMatch(hashValue -> (!hashRefLine.containsKey(hashValue)
```

Comment:
Re throwing an exception - we could do that if we wanted to merge the parsing code into master, but I don't think we will do that before the rest of the derivations code is done.

Re streaming twice - that method was already not very nice in that it does two things at the same time ...